File: crates/algosketch-core/tests/fixtures/binary_search.py
def binary_search(items, target):
    low: int = 0
    high: int = len(items) - 1
    while low <= high:
        mid: int = (low + high) // 2
        if items[mid] == target:
            return mid
        elif items[mid] < target:
            low = mid + 1
        else:
            high = mid - 1
    return -1

File: crates/algosketch-core/tests/fixtures/binary_search.cpp
int binary_search(int items[], int target, int size) {
    int low = 0;
    int high = size - 1;
    while (low <= high) {
        int mid = (low + high) / 2;
        if (items[mid] == target) {
            return mid;
        } else if (items[mid] < target) {
            low = mid + 1;
        } else {
            high = mid - 1;
        }
    }
    return -1;
}
